internal/api/web/views/components/quiz_components/answer_button.templ

```diff
@@ -0,0 +1,55 @@
+package quiz_components
+
+import "fmt"
+import "github.com/google/uuid"
+
+type OptionalABTN struct {
+	selected bool
+	disabled bool
+}
+
+type OptionABTN func(*OptionalABTN) OptionalABTN
+
+//Selects the button
+func SelectedAButton() OptionABTN {
+	return func(o *OptionalABTN) OptionalABTN {
+		o.selected = true
+		return *o
+	}
+}
+
+//Disables the button
+func DisabledAButton() OptionABTN {
+	return func(o *OptionalABTN) OptionalABTN {
+		o.disabled = true
+		return *o
+	}
+}
+
+//Assigns the correct style to the button based upon the options
+func assignCorrectStyle(option ...OptionABTN) string {
+	style := ""
+	for _, o := range option {
+		if o(&OptionalABTN{}).selected {
+			style += "border-4 border-blue-500 border-solid"
+		}
+	}
+	return style
+}
+
+//Checks if the button should be disabled based upon the options
+func checkIfIsDisabled(option ...OptionABTN) bool {
+	for _, o := range option {
+		if o(&OptionalABTN{}).disabled {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Still a bit confused about why the need for complex options here? Could maybe have been simplified?

Reply:
please look at already answered comments before making me need to repeat myself multiple times. But to say it shortly, to avoid creating duplicates, and forcing people to set values that they might not know (and therefore avoiding using intrusive default values).